### ISSUE TITLE
parser:  ':?' in type annotation does not produce a syntax error

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#5953]
Swapping the question mark and colon in a type annotation should produce a syntax error but it does not

Both lines in the snippet below should be incorrect syntax, but the snippet is parsed successfully

* [Oxc Playground](https://playground.oxc.rs/#eNpNTkFuxDAI/AriHCnq1VW1t74iF8dLKqs2uOB0E63y9zpqvV1xgIFhZu4Y0OE4QuQgqhSqgyBJePxayWoUhuz1E+zmS6Hr1FhsFTZ3AV7zTApv8PI68cRNg7aqlKl1n0uiTt7BweVRT284oKC7o658Ntu5+g1d1ZUGTJErusUna8CCFOoX2/MsqaOqnm0RzX/kY8Di1UibZJtPmT4/qH0R5EofdJo3kCPHJf6fuKqk9yS3M9s36SzWIvx6HMcPVadseQ==)
* [TypeScript Playground](https://www.typescriptlang.org/play/?#code/PTAEEsDsGMHsCd4FNoBcBco4BtaWAI4CuSAzquHqALYCG8A1qKQO60AO7SAJgFByRyoAB7oA-KEhFqAIyTxQAXlABGANy9eIUEmGpk1JDuG1q7bEn54hAT1CYJj0E6mz5S1WqA)

```ts
// incorrect: colon/question mark swapped
const x:? number = 1;

// extreme example
const y : ? ? ? ? ? number = 1;
```

**Comment by Boshen** [comment]
The AST is correct. Expect we need to produce an error:

```
    function checkJSDocTypeIsInJsFile(node: Node): void {
        if (!isInJSFile(node)) {
            if (isJSDocNonNullableType(node) || isJSDocNullableType(node)) {
                const token = tokenToString(isJSDocNonNullableType(node) ? SyntaxKind.ExclamationToken : SyntaxKind.QuestionToken);
                const diagnostic = node.postfix
                    ? Diagnostics._0_at_the_end_of_a_type_is_not_valid_TypeScript_syntax_Did_you_mean_to_write_1
                    : Diagnostics._0_at_the_start_of_a_type_is_not_valid_TypeScript_syntax_Did_you_mean_to_write_1;
                const typeNode = node.type;
                const type = getTypeFromTypeNode(typeNode);
                grammarErrorOnNode(
                    node,
                    diagnostic,
                    token,
                    typeToString(
                        isJSDocNullableType(node) && !(type === neverType || type === voidType)
                            ? getUnionType(append([type, undefinedType], node.postfix ? undefined : nullType)) : type,
                    ),
                );
            }
            else {
                grammarErrorOnNode(node, Diagnostics.JSDoc_types_can_only_be_used_inside_documentation_comments);
            }
        }
    }
```
